### PR TITLE
docs: update comparison table for node-fetch@3

### PR DESCRIPTION
**Pull request by ziv**
#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [x] I have included some tests. (no tests needed)
- [x] If it's a new feature, I have included documentation updates in both the README and the types.

No code changes, not tests added.
This PR is update the comparison table to add `node-fetch` version 3 new features.

1. node-fetch support for HTTP2
2. node-fetch support  Advanced timeouts using AbortController

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -170,7 +170,7 @@ For advanced JSON usage, check out the [`parseJson`](documentation/2-options.md#
 
 |                       | `got`               | [`request`][r0]    | [`node-fetch`][n0]   | [`ky`][k0]               | [`axios`][a0]      | [`superagent`][s0]     |
 |-----------------------|:-------------------:|:------------------:|:--------------------:|:------------------------:|:------------------:|:----------------------:|
-| HTTP/2 support        | :heavy_check_mark:¹ | :x:                | :x:                  | :x:                      | :x:                | :heavy_check_mark:\*\* |
+| HTTP/2 support        | :heavy_check_mark:¹ | :x:                | :heavy_check_mark:   | :x:                      | :x:                | :heavy_check_mark:\*\* |
```
Comment (szmarczak) [inline readme.md:173]: AFAIK it doesn't support HTTP/2 yet

https://github.com/node-fetch/node-fetch/issues/342

**Comment by sindresorhus** [comment]
Please link to sources that confirm your assertions.

File: readme.md [hunk 2]
```diff
@@ -182,7 +182,7 @@ For advanced JSON usage, check out the [`parseJson`](documentation/2-options.md#
 | Retries on failure    | :heavy_check_mark:  | :x:                | :x:                  | :heavy_check_mark:       | :x:                | :heavy_check_mark:     |
 | Progress events       | :heavy_check_mark:  | :x:                | :x:                  | :heavy_check_mark:\*\*\* | Browser only       | :heavy_check_mark:     |
 | Handles gzip/deflate  | :heavy_check_mark:  | :heavy_check_mark: | :heavy_check_mark:   | :heavy_check_mark:       | :heavy_check_mark: | :heavy_check_mark:     |
-| Advanced timeouts     | :heavy_check_mark:  | :x:                | :x:                  | :x:                      | :x:                | :x:                    |
+| Advanced timeouts     | :heavy_check_mark:  | :x:                | :heavy_check_mark:   | :x:                      | :x:                | :x:                    |
```
Comment (szmarczak) [inline readme.md:185]: I can't find any advanced timeouts usage. Can you provide a reference please?

File: readme.md (hunk 1 above)
Comment (ziv) [reply readme.md]: sorry, replied for the wrong comment :(

File: readme.md (hunk 2 above)
Comment (ziv) [reply readme.md:185]: https://github.com/node-fetch/node-fetch#request-cancellation-with-abortsignal

Comment (sindresorhus) [reply readme.md:185]: That's just normal timeout which all HTTP kind support, as far as I can see.

Comment (ziv) [reply readme.md:185]: I'll close this PR

**Pull request closed without merging**